crates/aggchain-proof-core/src/bridge/mod.rs

```diff
@@ -651,39 +650,38 @@ mod tests {
             })
             .collect();
 
-        // Instantiate the HostExecutor for the prev and new L2 blocks
+        // Instantiate the EvmSketch for the prev and new L2 blocks
         let (prev_l2_block_executor, new_l2_block_executor) = {
             let rpc_url_l2 = std::env::var(format!("RPC_{chain_id_l2}"))
                 .expect("RPC URL must be defined")
                 .parse::<Url>()
                 .expect("Invalid URL format");
 
-            let provider_l2: RootProvider<alloy::network::AnyNetwork> =
-                RootProvider::new_http(rpc_url_l2.clone());
+            let prev = EvmSketch::builder()
+                .at_block(BlockNumberOrTag::Number(initial_block_number))
+                .with_genesis(Genesis::Sepolia)
+                .el_rpc_url(rpc_url_l2.clone())
+                .build()
+                .await?;
 
-            let prev = HostExecutor::new(
-                provider_l2.clone(),
-                BlockNumberOrTag::Number(initial_block_number),
-            )
-            .await?;
-
-            let new = HostExecutor::new(
-                provider_l2.clone(),
-                BlockNumberOrTag::Number(final_block_number),
-            )
-            .await?;
+            let new = EvmSketch::builder()
+                .at_block(BlockNumberOrTag::Number(final_block_number))
+                .with_genesis(Genesis::Sepolia)
+                .el_rpc_url(rpc_url_l2.clone())
+                .build()
+                .await?;
```

Comment:
You could do something like that too:

```suggestion
            let sketch = EvmSketch::builder()
                .with_genesis(Genesis::Sepolia)
                .el_rpc_url(rpc_url_l2.clone());
           let prev = sketch.clone()
           	.at_block(BlockNumberOrTag::Number(initial_block_number))
                .build()
                .await?;
            let new = sketch.clone()
           	.at_block(BlockNumberOrTag::Number(final_block_number))
                .build()
                .await?;
```